frontend/src/util/worksheet_utils.js

```diff
@@ -304,15 +304,16 @@ export function getAfterSortKey(item, subFocusIndex) {
     // The default after_sort_key is -1 when inserting an item on top of the worksheet,
     // so the item's sort_key should always be >= 0 (sort_key > after_sort_key)
     if (!item) return -1;
-    if (item.mode === 'image_block') {
-        // image_block stores the sort_key in a different property than do other blocks
+    if (item.mode === 'image_block' || item.mode === 'contents_block') {
+        // image_block & content_block stores the sort_key in a different property than do other blocks
         return item['bundles_spec']['bundle_infos'][0]['sort_key'];
     }
     const sort_keys = item.sort_keys || [];
     if (sort_keys[subFocusIndex] || sort_keys[subFocusIndex] === 0) {
         return sort_keys[subFocusIndex];
     }
-    return Math.max(...sort_keys);
+    const afterSortKey: number = Math.max(...sort_keys);
+    return isFinite(afterSortKey) ? afterSortKey : -1;
```

Comment:
What edge case are we checking for here? Is it just that afterSortKey is not a number -- and in that case, when does that happen? (in that case, we should probably use `isNumber` instead)

Reply:
we are checking for whether the afterSortKey is inf or not, so I think we better use `isFinite` to check. WIth inf afterSortKey, the frontend will report an error.

Reply:
When would afterSortKey be equal to inf?

Reply:
when some error occurs, you can find inf check in NewUpload as well.